tests/test_arrow_dataset.py

```diff
@@ -1948,6 +1948,27 @@ def test_concatenate_datasets_duplicate_columns(dataset):
     assert "duplicated" in str(excinfo.value)
 
 
+@pytest.mark.parametrize("in_memory", [False, True])
+def test_dataset_add_column(in_memory, dataset_dict, arrow_path):
```

Comment:
If you want to test the cases where a dataset is formatted / has an indices mapping / has classlabels, feel free to check out how I did it in #2254 for the `add_item` test